src/diart/models.py

```diff
@@ -7,29 +7,47 @@
 from requests import HTTPError
 
 try:
-    from pyannote.audio import Inference, Model
-    from pyannote.audio.pipelines.speaker_verification import (
-        PretrainedSpeakerEmbedding,
-    )
+    import pyannote.audio.pipelines.utils as pyannote_loader
+    from pyannote.audio.utils.powerset import Powerset
 
     _has_pyannote = True
 except ImportError:
     _has_pyannote = False
 
 
+class PowersetAdapter(nn.Module):
+    def __init__(self, segmentation_model: nn.Module):
+        super().__init__()
+        self.model = segmentation_model
+        specs = self.model.specifications
+        max_speakers_per_frame = specs.powerset_max_classes
+        max_speakers_per_chunk = len(specs.classes)
+        self.powerset = Powerset(max_speakers_per_chunk, max_speakers_per_frame)
+
+    @property
+    def specifications(self):
+        return self.model.specifications
+
+    @property
+    def audio(self):
+        return self.model.audio
+
+    def forward(self, waveform: torch.Tensor) -> torch.Tensor:
+        return self.powerset.to_multilabel(self.model(waveform), soft=False)
+
+
 class PyannoteLoader:
     def __init__(self, model_info, hf_token: Union[Text, bool, None] = True):
         super().__init__()
         self.model_info = model_info
         self.hf_token = hf_token
 
-    def __call__(self) -> Callable:
-        try:
-            return Model.from_pretrained(self.model_info, use_auth_token=self.hf_token)
-        except HTTPError:
-            return PretrainedSpeakerEmbedding(
-                self.model_info, use_auth_token=self.hf_token
-            )
+    def __call__(self) -> nn.Module:
```

Comment:
I think the rebasing broke something here but the solution with the `HTTPError` was not very clean either. Is there anything better though?

Reply:
I agree that the HTTPError wasn't the best solution but I'm not aware of anything better we can do. That said, something definitely went wrong during the rebase